src/twig.core.js

```diff
@@ -960,10 +968,18 @@ var Twig = (function (Twig) {
                         params.url = location;
                         params.data = data;
 
-                        template = new Twig.Template(params);
+                        if (fetchTemplateSource === true) {
+                            templateSource = data;
```

Comment:
Why are you using a separate `templateSource` variable? If you just use `template`, you don't need to change the callbacks resp. return below ...

```js
                        if (fetchTemplateSource === true) {
                            template = data;
                        } else {
                            template = new Twig.Template(params);
                        }
```